src/cplscheme/BaseCouplingScheme.cpp

```diff
@@ -104,35 +104,44 @@ void BaseCouplingScheme::sendData(const m2n::PtrM2N &m2n, const DataMap &sendDat
   for (const auto &data : sendData | boost::adaptors::map_values) {
     const auto stamples = data->stamples();
     PRECICE_ASSERT(stamples.size() > 0);
-    data->sample() = stamples.back().sample;
+
+    auto serializedValues = data->getSerializedValues();
```

Comment:
This will copy a bunch of data. You could simplify this by creating a `std::vector<double> serializedValues` outside the loop.

Then you pass this as `std::vector<double>&` to:
```suggestion
    data->serializeValuesTo(serializedValues);
```

Inside you would call `.resize()`.
This should save you allocations.

Reply:
See comment above. I'll follow the pattern provided by `SerializedMesh` etc. This should also have the same effect.